widgetastic_manageiq/__init__.py

```diff
@@ -4437,3 +4437,150 @@ def fill(self, value):
+
+    def expand_map_associated_plans(self, map_name):
+        """Click on associated plans to expand to show details."""
+        try:
+            el = self._get_map_element(map_name)
+            self.browser.click(self.browser.element(
+                self.ITEM_ASSOCIATED_PLANS_BUTTON_LOCATOR, parent=el))
+        except NoSuchElementException:
+            raise ItemNotFound("There are no associated plans with map {}".format(map_name))
+
+    def read(self):
+        return self.all_items
+
+    def get_map_status(self, map_name):
+        el = self._get_map_element(map_name)
+        return self.browser.text(self.browser.element(self.ITEM_STATUS_LOCATOR,
+             parent=el))
+
+    def _get_sources(self, map_name):
+        el = self._get_map_element(map_name)
+        for item in self.browser.elements(self.SOURCE_ITEMS_LIST_LOCATOR, parent=el):
+            return self.browser.text(item)
+
+    def _get_targets(self, map_name):
+        el = self._get_map_element(map_name)
+        for item in self.browser.elements(self.TARGET_ITEMS_LIST_LOCATOR, parent=el):
+            return self.browser.text(item)
+
+    def get_map_source_clusters(self, map_name):
+        if not self.is_clusters_expanded(map_name):
+            self.expand_map_clusters(map_name)
+        return self._get_sources(map_name)
+
+    def get_map_target_clusters(self, map_name):
+        if not self.is_clusters_expanded(map_name):
+            self.expand_map_clusters(map_name)
+        return self._get_targets(map_name)
+
+    def get_map_source_networks(self, map_name):
+        if not self.is_networks_expanded(map_name):
+            self.expand_map_networks(map_name)
+        return self._get_sources(map_name)
+
+    def get_map_target_networks(self, map_name):
+        if not self.is_networks_expanded(map_name):
+            self.expand_map_networks(map_name)
+        return self._get_targets(map_name)
+
+    def get_map_source_datastores(self, map_name):
+        if not self.is_datastores_expanded(map_name):
+            self.expand_map_datastores(map_name)
+        return self._get_sources(map_name)
+
+    def get_map_target_datastores(self, map_name):
+        if not self.is_datastores_expanded(map_name):
+            self.expand_map_datastores(map_name)
+        return self._get_targets(map_name)
+
+    def get_associated_plans_count(self, map_name):
+        try:
+            el = self._get_map_element(map_name)
+            return self.browser.text(self.browser.element(self.ITEM_ASSOCIATED_PLANS_BUTTON_LOCATOR,
+                 parent=el))
+        except NoSuchElementException:
+            # Plan with no associated plans
+            return None
+
+    def get_associated_plans(self, map_name):
+        if not self.is_associated_plans_expanded(map_name):
+            self.expand_map_associated_plans(map_name)
+        el = self._get_map_element(map_name)
+        return self.browser.text(self.browser.element(self.ITEM_ASSOCIATED_PLANS_LIST_LOCATOR,
```

Comment:
Feels a bit strange that this isn't returning a list of names, since its plural? If there's some reason why it doesn't or can't, please describe the behavior and what you expect this element to locate in the docblock.

Reply:
We cannot split it to return a list as there is no way to differentiate, its all single block of text under a div. I have asked them if they can divide those in different divs/li tags, if they decide to do so, we can revisit this and update the widget. For now, it just returns a string.